src/order/rel_classes.lean

```diff
@@ -76,6 +76,24 @@ protected theorem is_asymm.is_irrefl [is_asymm α r] : is_irrefl α r :=
 protected theorem is_total.is_trichotomous (r) [is_total α r] : is_trichotomous α r :=
 ⟨λ a b, or.left_comm.1 (or.inr $ total_of r a b)⟩
 
+/-- Note this cannot be an instance as it would loop with itself. -/
+theorem is_refl_swap {α : Type u} (r : α → α → Prop) [is_refl α r] :
```

Comment:
We already have `is_refl.swap` at line 49 of the same file. Same for `is_irrefl.swap`.

Reply:
Nice catch! Should I rename the other two theorems to use dot notation to match with these?